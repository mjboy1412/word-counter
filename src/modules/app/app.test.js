import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "jest-styled-components";

import App from "./index";
import {
  CHARACTERS_LIMITATION,
  STRING_BELOW_40_CHAR,
  LENGTH_OF_STRING_BELOW_40_CHAR,
  STRING_40_TO_50_CHAR,
  LENGTH_OF_STRING_40_TO_50_CHAR,
  STRING_BEYONG_50_CHAR,
  LENGTH_OF_STRING_BEYONG_50_CHAR,
  WORD_COUNTER_COLOR,
} from "./contants";

it("snapshot testing", () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

it("test word-counter when it is below 40", () => {
  const { getByTestId } = render(<App />);
  fireEvent.change(getByTestId(/textarea/g), {
    target: { value: STRING_BELOW_40_CHAR },
  });
  expect(getByTestId(/charCounter/g)).toHaveTextContent(
    CHARACTERS_LIMITATION - LENGTH_OF_STRING_BELOW_40_CHAR
  );
});

it("test word-counter when it is below 40", () => {
  const { getByTestId } = render(<App />);
  fireEvent.change(getByTestId(/textarea/g), {
    target: { value: STRING_BELOW_40_CHAR },
  });
  expect(getByTestId(/charCounter/g)).toHaveTextContent(
    CHARACTERS_LIMITATION - LENGTH_OF_STRING_BELOW_40_CHAR
  );
  expect(getByTestId(/charCounter/g)).toHaveStyleRule(
    "color",
    WORD_COUNTER_COLOR.BLUE
  );
});

it("test word-counter when it is between 40 and 50", () => {
  const { getByTestId } = render(<App />);
  fireEvent.change(getByTestId(/textarea/g), {
    target: { value: STRING_40_TO_50_CHAR },
  });
  expect(getByTestId(/charCounter/g)).toHaveTextContent(
    CHARACTERS_LIMITATION - LENGTH_OF_STRING_40_TO_50_CHAR
  );
  expect(getByTestId(/charCounter/g)).toHaveStyleRule(
    "color",
    WORD_COUNTER_COLOR.YELLOW
  );
});

it("test word-counter when it is below 40", () => {
  const { getByTestId } = render(<App />);
  fireEvent.change(getByTestId(/textarea/g), {
    target: { value: STRING_BEYONG_50_CHAR },
  });
  expect(getByTestId(/charCounter/g)).toHaveTextContent(
    CHARACTERS_LIMITATION - LENGTH_OF_STRING_BEYONG_50_CHAR
  );
  expect(getByTestId(/charCounter/g)).toHaveStyleRule(
    "color",
    WORD_COUNTER_COLOR.RED
  );
});
