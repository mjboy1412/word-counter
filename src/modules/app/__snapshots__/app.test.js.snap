// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<DocumentFragment>
  .c0 {
  height: 700px;
  width: 700px;
  border: 1px solid black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 80px;
  padding: 25px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  max-height: 80%;
  border: none;
  outline: none;
  resize: none;
  width: 100%;
  padding: 25px;
  box-sizing: border-box;
  font-size: 16px;
  font-family: Segoe UI;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
}

.c5 {
  height: 57px;
  border-top: 1px solid #ccc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  font-size: 42px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c3 {
  background: rgba(29,161,242,1);
  color: #fff;
  border-radius: 30px;
  border: none;
  font-size: 18px;
  outline: none;
  width: 94px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c6 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  font-size: 30px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c8 {
  font-size: 30px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c9 {
  font-size: 30px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c10 {
  font-size: 30px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c11 {
  min-width: 77px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 {
  font-size: 30px;
  color: rgba(29,161,242,1);
  cursor: pointer;
}

.c12 {
  border-radius: 1.5rem;
  border: 2px solid rgba(29,161,242,1);
  color: rgba(29,161,242,1);
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3px 5px;
  height: 1em;
}

.c13 {
  width: 0px;
  border-left: 1px solid #ccc;
  content: "";
  height: 20px;
  margin: 0px 10px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        class="c2"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.854 4.146a.5.5 0 010 .708l-7 7a.5.5 0 01-.708-.708l7-7a.5.5 0 01.708 0z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M4.146 4.146a.5.5 0 000 .708l7 7a.5.5 0 00.708-.708l-7-7a.5.5 0 00-.708 0z"
          fill-rule="evenodd"
        />
      </svg>
      <button
        class="c3"
      >
        Tweet
      </button>
    </div>
    <textarea
      class="c4"
      data-testid="textarea"
      placeholder="Type something here please !!"
    />
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <svg
          class="c7"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.002 2h-12a1 1 0 00-1 1v10a1 1 0 001 1h12a1 1 0 001-1V3a1 1 0 00-1-1zm-12-1a2 2 0 00-2 2v10a2 2 0 002 2h12a2 2 0 002-2V3a2 2 0 00-2-2h-12z"
            fill-rule="evenodd"
          />
          <path
            d="M10.648 7.646a.5.5 0 01.577-.093L15.002 9.5V14h-14v-2l2.646-2.354a.5.5 0 01.63-.062l2.66 1.773 3.71-3.71z"
          />
          <path
            clip-rule="evenodd"
            d="M4.502 7a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="c8"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          t="1569683725598"
          version="1.1"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M551.5 490.5H521c-4.6 0-8.4 3.7-8.4 8.4V720c0 4.6 3.7 8.4 8.4 8.4h30.5c4.6 0 8.4-3.7 8.4-8.4V498.9c-0.1-4.6-3.8-8.4-8.4-8.4zM477.3 600h-88.1c-4.6 0-8.4 3.7-8.4 8.4v23.8c0 4.6 3.7 8.4 8.4 8.4h47.6v0.7c-0.6 29.9-23 49.8-56.5 49.8-39.2 0-63.6-30.7-63.6-81.4 0-50.1 23.9-80.6 62.3-80.6 28.1 0 47.5 13.5 55.4 38.3l0.9 2.8h49.2l-0.7-4.6C475.9 515.9 434.7 484 379 484c-68.8 0-113 49.4-113 125.9 0 77.5 43.7 126.1 113.6 126.1 64.4 0 106-40.3 106-102.9v-24.8c0-4.6-3.7-8.3-8.3-8.3z"
          />
          <path
            d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM602 137.8L790.2 326H602V137.8zM792 888H232V136h302v216c0 23.2 18.8 42 42 42h216v494z"
          />
          <path
            d="M608.2 727.8h32.3c4.6 0 8.4-3.7 8.4-8.4v-84.8h87.8c4.6 0 8.4-3.7 8.4-8.4v-25.5c0-4.6-3.7-8.4-8.4-8.4h-87.8v-58.9h96.8c4.6 0 8.4-3.7 8.4-8.4v-26.8c0-4.6-3.7-8.4-8.4-8.4H608.2c-4.6 0-8.4 3.7-8.4 8.4v221.1c0 4.8 3.8 8.5 8.4 8.5z"
          />
        </svg>
        <svg
          class="c9"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2 12.5a.5.5 0 01.5-.5h7a.5.5 0 010 1h-7a.5.5 0 01-.5-.5zm0-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5zm0-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5zm0-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5z"
            fill-rule="evenodd"
          />
        </svg>
        <svg
          class="c10"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 16s6-5.686 6-10A6 6 0 002 6c0 4.314 6 10 6 10zm0-7a3 3 0 100-6 3 3 0 000 6z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="c11"
      >
        <span
          class="c12"
          color="rgba(29,161,242,1)"
          data-testid="charCounter"
          title="remaining characters"
        >
          50
        </span>
         
        <div
          class="c13"
        />
        <svg
          class="c14"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16 8A8 8 0 110 8a8 8 0 0116 0zM8.5 4a.5.5 0 00-1 0v3.5H4a.5.5 0 000 1h3.5V12a.5.5 0 001 0V8.5H12a.5.5 0 000-1H8.5V4z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
